### PR TITLE
Remove "beta" warning for Units catalog [POFSP-195]

### PR DESCRIPTION
**Pull request by polomani**
## Description
- removed beta warning for the unit catalog
- updated refs to API documentation (links will work a bit later https://github.com/cognitedata/service-contracts/pull/2375)

### CODE REVIEW

File: cognite/client/_api/units.py [hunk 1]
```diff
@@ -41,7 +38,7 @@ def retrieve(self, external_id: MutableSequence[str], ignore_unknown_ids: bool =
     def retrieve(
         self, external_id: str | MutableSequence[str], ignore_unknown_ids: bool = False
     ) -> Unit | UnitList | None:
-        """`Retrieve one or more unit <https://pr-50.units-api.preview.cogniteapp.com/#tag/Units/operation/retrieve_units_by_ids_api_v1_projects__project__units_byids_post>`_
+        """`Retrieve one or more unit <https://developer.cognite.com/api#tag/Units/operation/byIdsUnits>`_
```
Comment (haakonvt) [inline cognite/client/_api/units.py:41]: None of these urls seem to work yet?

Comment (polomani) [reply cognite/client/_api/units.py:41]: no, but they will be there once we merge https://github.com/cognitedata/service-contracts/pull/2375

File: cognite/client/_api/units.py [hunk 2]
```diff
@@ -98,25 +95,22 @@ def __init__(
         config: ClientConfig,
         api_version: str | None,
         cognite_client: CogniteClient,
-        warning: FeaturePreviewWarning,
     ) -> None:
         super().__init__(config, api_version, cognite_client)
```
Comment (haakonvt) [inline cognite/client/_api/units.py:99]: You can remove this entire init method now.

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,6 +17,10 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [7.4.0] - 2023-11-24
+### Added
+- Unit Catalog API is now considered stable and can be accessed at `client.units`.
```
Comment (haakonvt) [inline CHANGELOG.md:22]:
```suggestion
## [7.4.0] - 2023-11-24
### Changed
- Unit Catalog API is out of beta and will no longer issue warnings on usage. Access is unchanged: `client.units`.
```

File: cognite/client/_api/units.py (hunk 1 above)
Comment (polomani) [reply cognite/client/_api/units.py:41]: now they work @haakonvt

**Review by haakonvt — APPROVED** [review]
lgtm, two x nit

**Pull request merged**